product-service/handlers/getProductsById.ts

```diff
@@ -1,20 +1,27 @@
 import { APIGatewayProxyHandler } from 'aws-lambda';
 import 'source-map-support/register';
+import { Client } from 'pg';
+
+import { dbOptions } from '../db-options';
+import { PGTransaction } from '../utils';
 
-import axios from "axios";
-import { DATABASE_ENDPOINT } from "../utils";
 
 export const getProductsById: APIGatewayProxyHandler = async (event, _context) => {
+  console.log('pathParameters @getProductsById: ', event.pathParameters);
+
+  const client = new Client(dbOptions);
+  await client.connect();
+
   try {
+    await client.query(PGTransaction.begin);
     const { productId } = event.pathParameters;
-    const response = await axios.get(`${ DATABASE_ENDPOINT }/products`);
-    const { data: products } = response;
-    let product;
-
-    if (products && products.length) {
-      product = products.find(product => product.id === Number(productId));
-    }
+    const { rows: products } = await client.query(`
+      SELECT id, count, price, title, description FROM products
+      INNER JOIN stocks s ON (products.id = s.product_id AND s.product_id = $1)
+    `, [productId]);
```

Comment:
You should check whether the productId matches the UUID format to prevent HTTP 500 errors on requests like

GET https://f6hqzu6u08.execute-api.eu-west-1.amazonaws.com/dev/products/3

If productId is not matches UUID is seems to be good to respond with HTTP 400 error.

Reply:
will do, thanks!

Reply:
done